NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -125,8 +145,11 @@ class NeuroID
             }
 
             // set call activity listener
-            nidCallActivityListener = NIDCallActivityListener(dataStore, VersionChecker())
-            this.getApplicationContext()?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            if (nidSDKConfig.callInProgress) {
+                nidCallActivityListener = NIDCallActivityListener(dataStore, VersionChecker())
+                this.getApplicationContext()
+                    ?.let { nidCallActivityListener?.setCallActivityListener(it) }
+            }
```

Comment:
is there a race condition between this and loading the config? Would it be better to turn the config service into an async/await so we can get the config before moving forward?

Reply:
Good point. I think a blocking approach here might be appropriate to avoid a setting race condition.